app/services/ClaimToAdjustService.scala

```diff
@@ -92,42 +98,37 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
   }
 
   private def checkCrystallisation(nino: Nino, taxYearList: List[TaxYear])(implicit hc: HeaderCarrier): Future[Option[TaxYear]] = {
-    taxYearList match {
-      case ::(head, next) => isTaxYearNonCrystallised(head, nino).flatMap {
-        case true => Future.successful(Some(head))
-        case false => checkCrystallisation(nino, next)
+    taxYearList.foldLeft(Future.successful(Option.empty[TaxYear])) { (acc, item) =>
+      acc.flatMap {
+        case Some(_) => acc
+        case None => isTaxYearNonCrystallised(item, nino) map {
+          case true => Some(item)
+          case false => None
+        }
       }
-      case Nil => Future.successful(None)
     }
   }
 
-  // Next 2 methods taken from Calculation List Service (with small changes), but I don't like the code duplication. Is there any solution to this?
-  private def isTaxYearNonCrystallised(taxYear: TaxYear, nino: Nino)(implicit hc: HeaderCarrier): Future[Boolean] = {
-
+  private def isFutureTaxYear(taxYear: TaxYear): Boolean = {
```

Comment:
this method is good candidate to be moved into TaxYear case class;
something like that would work well:

` def isFutureTaxYear(implicit dateService: DateService): Boolean = ???`